File: src/Java/401-500/415.AddStrings.java

import java.awt.List;
import java.lang.annotation.Retention;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;

class Solution {
    public String addStrings(String num1, String num2) {
        int n1 = num1.length();
        int n2 = num2.length();
        int n = n1;
        int c = 0;
        String num = "";
        if (n1 < n2) {
            n = n2;
            for (int i = 0; i < n2 - n1; i++) {
                num1 = "0" + num1;
            }
        } else {
            n = n1;
            for (int i = 0; i < n1 - n2; i++) {
                num2 = "0" + num2;
            }
        }
        for (int i = 0; i < n; i++) {
            int j = n - 1 - i;
            int bit = num1.charAt(j) - '0' + num2.charAt(j) - '0' + c;
            if (bit >= 10) {
                c = 1;
                bit -= 10;
            } else {
                c = 0;
            }
            num = (char)(bit + '0') + num;
        }
        if (c == 1) {
            num = (char)(c + '0') + num;
        }
        return num;
    }
}

    

File: src/Python/401-500/415.AddStrings.py

class Solution:
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        n1 = len(num1)
        n2 = len(num2)
        if n1 < n2:
            num1 = '0' * (n2 - n1) + num1
            n = n2
        else:
            num2 = '0' * (n1 - n2) + num2
            n = n1
        num = []
        c = 0
        for i in range(n):
            j = n - 1 - i
            bit = int(num1[j]) + int(num2[j]) + c       
            if bit >= 10:
                bit -= 10
                c = 1 
            else:
                c = 0
            num.append(str(bit))
        if c == 1:
            num.append('1')
        num = num[::-1]
        return ''.join(num)

if __name__ == '__main__':
    solution = Solution()
    print(solution.addStrings('123', '32111'))
else:
    pass
